tox.ini

```diff
@@ -13,7 +13,7 @@ extras =
 deps =
     pipprevious: pip==22.2.*
     piplatest: pip
-    pipmain: -e git+https://github.com/pypa/pip.git@main#egg=pip
+    pipmain: https://github.com/pypa/pip/archive/main.zip
```

Comment:
That looks interesting! I never knew this was possible. What are the benefits? Will it install newer pip when something new is updated in pip? Will it avoid reinstalling pip when the archive is not changed? Any side-effects on packaging?

-- Mainly asking because I want to know if I should use the same approach in my projects.

Reply:
> I never knew this was possible.

I've [seen](https://sourcegraph.com/search?q=context:global+file:%5Etox%5C.ini%24+%28master%7Cmain%29.zip&patternType=regexp&sm=1) this mostly in Django projects.

> What are the benefits?

Back in the day it was [faster](https://stackoverflow.com/a/24811490/1529164) than installing from git, especially with submodules. I've tested locally for pip-tools and it's slightly faster.

> Will it install newer pip when something new is updated in pip?

I've tested on my own project. It re-installs only if version is updated. I think that's okay for `tox` where you can always recreate environment using `tox --recreate` and get latest updates.

> Will it avoid reinstalling pip when the archive is not changed?

It won't be reinstalled If version is not updated.

> Any side-effects on packaging, versioning in particular?

Not that I know of.

---

Unfortunately I failed to research why pip fails to re-install itself from git on Windows. It's certainly a bug, but I spent too much time and gave up. I've tried to install the zip and it worked like a charm, hence this PR.

Reply:
@atugushev in your log it's visible that `-e` and the pip repo URL are parsed as a single argument while they should be separate.

Reply:
So it could also be a tox bug too.